toolkit/pallets/block-production-log/src/lib.rs

```diff
@@ -96,5 +112,15 @@ pub mod pallet {
 			Log::<T>::put(to_retain);
 			to_return
 		}
+
+		pub fn peek_prefix<'a>(slot: Slot) -> impl Iterator<Item = (Slot, T::BlockProducerId)> {
+			Log::<T>::get().into_iter().filter(move |(s, _)| s <= &slot)
```

Comment:
if we assume that our collection is sorted, then we can utilize `skip_while`:
```suggestion
			Log::<T>::get().into_iter().skip_while(move |(s, _)| s <= &slot)
```

Reply:
take_while*. changed